neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -93,11 +91,15 @@ async def call_tool(self, request_id, metadata: Dict[str, Any], tool_name: str,
             # No such tool is found:
             return McpErrorsUtil.get_tool_error(request_id, f"Tool not found: {tool_name}")
         service: AsyncAgentService = service_provider.get_service()
+        tool_timeout_seconds: float = service.get_request_timeout_seconds()
+        if tool_timeout_seconds <= 0.0:
+            # For asyncio.timeout(), None means no timeout:
+            tool_timeout_seconds = None
 
```

Comment:
As for neuro-san chat request handler, get timeout value from agent service instance.